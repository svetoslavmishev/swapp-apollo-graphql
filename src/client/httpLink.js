import { createHttpLink } from 'apollo-link-http';

const httpLink = createHttpLink({
  uri: process.env.REACT_APP_SWAPP_ENDPOINT
});

export default httpLink;
