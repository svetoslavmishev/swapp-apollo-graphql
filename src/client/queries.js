import gql from 'graphql-tag';

export const AUTH = gql`
  query AuthQuery {
    isAuthenticated @client
  }
`;

export const SIGN_IN = gql`
  mutation SignIn($email: String!, $password: String!) {
    signIn(email: $email, password: $password) {
      token
    }
  }
`;
