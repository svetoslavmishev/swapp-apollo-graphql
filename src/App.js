import React, { useState } from "react";
import "./App.css";
import { ApolloProvider } from "@apollo/react-hooks";
import { ThemeProvider } from "@material-ui/core/styles";

import Routes from "./routes/routes";
import client from "./client";
import { ThemeContext, themes } from "./themeContext";

function App() {
  const [currentTheme, setCurrentTheme] = useState("light");

  const toggleTheme = () => {
    setCurrentTheme(currentTheme === "light" ? "dark" : "light");
  };

  return (
    <ThemeContext.Provider value={{ currentTheme, toggleTheme }}>
      <ThemeProvider theme={themes}>
        <ApolloProvider client={client}>
          <Routes />
        </ApolloProvider>
      </ThemeProvider>
    </ThemeContext.Provider>
  );
}

export default App;
