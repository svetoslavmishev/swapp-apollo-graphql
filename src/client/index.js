import { ApolloClient } from 'apollo-client';
import { ApolloLink } from 'apollo-link';
import { InMemoryCache } from 'apollo-cache-inmemory';

import authLink from './authLink';
import httpLink from './httpLink';
import typeDefs from './local';

const cache = new InMemoryCache();
const client = new ApolloClient({
  cache,
  link: ApolloLink.from([authLink, httpLink]),
  typeDefs
});

cache.writeData({
  data: {
    isAuthenticated: !!localStorage.getItem('token')
  }
});

export default client;
