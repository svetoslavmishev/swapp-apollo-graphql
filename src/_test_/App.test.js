import React from 'react';
import { shallow } from 'enzyme';
import App from '../App';
import { ApolloProvider } from '@apollo/react-hooks';
import client from '../client';

describe('<App />', () => {
  it('renders <App /> without error', () => {
    const wrapper = shallow(
      <ApolloProvider client={client}>
        <App />
      </ApolloProvider>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
