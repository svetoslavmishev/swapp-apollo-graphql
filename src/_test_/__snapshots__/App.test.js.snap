// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders <App /> without error 1`] = `ShallowWrapper {}`;
